app/src/main/java/org/greenstand/android/TreeTracker/preferences/Preferences.kt

```diff
@@ -1,71 +1,65 @@
-package org.greenstand.android.TreeTracker.managers
+package org.greenstand.android.TreeTracker.preferences
 
 import android.content.SharedPreferences
 
-object PrefKeys {
-
-    val ROOT = PrefKey("greenstand")
-
-    val SESSION = PrefKey("session")
+class Preferences(
+    private val prefs: SharedPreferences
+) {
 
-    val USER_SETTINGS = ROOT + PrefKey("user-settings")
-}
+    private var _planterInfoId: Long? = null
 
-open class PrefKey(val path: String) {
 
-    operator fun plus(prefKey: PrefKey): PrefKey {
-        return when (prefKey) {
-            is UserPrefKey -> UserPrefKey("$path/${prefKey.path}")
-            else -> PrefKey("$path/${prefKey.path}")
-        }
-    }
-
-    fun asUserPref(): UserPrefKey {
-        return UserPrefKey(path)
+    fun setPlanterInfoId(planterInfoId: Long?) {
+        _planterInfoId = planterInfoId
     }
-}
-
-class UserPrefKey(path: String) : PrefKey(path)
-
-class Preferences(
-    private val prefs: SharedPreferences,
-    private val userManager: UserManager
-) {
 
     fun getBoolean(prefKey: PrefKey, default: Boolean): Boolean {
         return prefs.getBoolean(computePath(prefKey), default)
     }
 
-    fun getString(prefKey: PrefKey, default: String): String {
+    fun getString(prefKey: PrefKey, default: String? = null): String? {
         return prefs.getString(computePath(prefKey), default)
     }
 
-    fun getFloat(prefKey: PrefKey, default: Float): Float {
+    fun getLong(prefKey: PrefKey, default: Long = -1): Long {
+        return prefs.getLong(computePath(prefKey), default)
+    }
+
+    fun getFloat(prefKey: PrefKey, default: Float = -1f): Float {
         return prefs.getFloat(computePath(prefKey), default)
     }
 
-    fun getInt(prefKey: PrefKey, default: Int): Int {
+    fun getInt(prefKey: PrefKey, default: Int = -1): Int {
         return prefs.getInt(computePath(prefKey), default)
     }
 
-    fun computePath(prefKey: PrefKey): String {
+    private fun computePath(prefKey: PrefKey): String {
         return when (prefKey) {
             is UserPrefKey -> {
-                prefKey.path + "/${userManager.planterInfoId}"
+                prefKey.path + "/${_planterInfoId}"
```

Comment:
It would be intuitive to have `_planterInfoId` within `UserPrefKey`, given it is associated with a specific `User`.

Reply:
If we do this, then we'd need to recreate all the keys every time someone logs out. Otherwise the new user would have UserPrefKeys with the previous users ID associated with them.